x/stake/client/rest/query.go

```diff
@@ -16,33 +19,97 @@ import (
 const storeName = "stake"
 
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *wire.Codec) {
+
+	// GET /stake/delegators/{delegatorAddr} // Get all delegations (delegation, undelegation and redelegation) from a delegator
 	r.HandleFunc(
-		"/stake/{delegator}/delegation/{validator}",
-		delegationHandlerFn(cliCtx, cdc),
+		"/stake/delegators/{delegatorAddr}",
+		delegatorHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+
+	// GET /stake/delegators/{delegatorAddr}/txs // Get all staking txs (i.e msgs) from a delegator
+	r.HandleFunc(
+		"/stake/delegators/{delegatorAddr}/txs",
+		delegatorTxsHandlerFn(cliCtx, cdc),
 	).Methods("GET")
 
+	// // GET /stake/delegators/{delegatorAddr}/validators // Query all validators that a delegator is bonded to
+	// r.HandleFunc(
+	// 	"/stake/delegators/{delegatorAddr}/validators",
+	// 	delegatorValidatorsHandlerFn(cliCtx, cdc),
+	// ).Methods("GET")
+
+	// GET /stake/delegators/{delegatorAddr}/validators/{validatorAddr} // Query a validator that a delegator is bonded to
+	// r.HandleFunc(
+	// 	"/stake/delegators/{delegatorAddr}/validators",
+	// 	delegatorValidatorHandlerFn(cliCtx, cdc),
+	// ).Methods("GET")
+
+	// GET /stake/delegators/{delegatorAddr}/delegations/{validatorAddr} // Query a delegation between a delegator and a validator
 	r.HandleFunc(
-		"/stake/{delegator}/ubd/{validator}",
-		ubdHandlerFn(cliCtx, cdc),
+		"/stake/delegators/{delegatorAddr}/delegations/{validatorAddr}",
+		delegationHandlerFn(cliCtx, cdc),
 	).Methods("GET")
 
+	// GET /stake/delegators/{delegatorAddr}/unbonding_delegations/{validatorAddr} // Query all unbonding_delegations between a delegator and a validator
 	r.HandleFunc(
-		"/stake/{delegator}/red/{validator_src}/{validator_dst}",
-		redHandlerFn(cliCtx, cdc),
+		"/stake/delegators/{delegatorAddr}/unbonding_delegations/{validatorAddr}",
+		unbondingDelegationsHandlerFn(cliCtx, cdc),
 	).Methods("GET")
 
+	/*
+			GET /stake/delegators/{addr}/validators/{addr}/txs // Get all txs to a validator performed by a delegator
+		 	GET /stake/delegators/{addr}/validators/{addr}/txs?type=bond // Get all bonding txs to a validator performed by a delegator
+			GET /stake/delegators/{addr}/validators/{addr}/txs?type=unbond // Get all unbonding txs to a validator performed by a delegator
+			GET /stake/delegators/{addr}/validators/{addr}/txs?type=redelegate // Get all redelegation txs to a validator performed by a delegator
+	*/
+	// r.HandleFunc(
+	// 	"/stake/delegators/{delegatorAddr}/validators/{validatorAddr}/txs",
+	// 	stakingTxsHandlerFn(cliCtx, cdc),
+	// ).Queries("type", "{type}").Methods("GET")
```

Comment:
Delete all commented out code in this file?

Reply:
Addressed